# 6. Delta Ladder

In [2]:
from fixedincome2025 import table

## Overview

* Recall that, given $D^{\$}$ and $C^{\$}$, we can estimate the change of the value of a bond portfolio for parallel yield curve shift
* As the yield curve doesn't always move parallelly, we need more granular control of the portfolio value for interest rate risk management
* The Delta ladder is a risk managing tool for this purpose

## Bump and Revaluate and Delta Ladder

* Having learned yield curve interpolation, we can obtain yield at any terms, and with that we can compute the value of arbitrary bonds/portfolios
* Today's curve is the base scenario, and the portfolio value computed with today's curve is the base PV (present value)
* Bumped curves:
    * There are 14 buckets on the below curve
    * Adding 1 bp to each bucket, we obtain 14 bumped curves
        * The first one has yields $\underline{4.24\%}, 4.18\%, 4.11\%, \ldots$
        * The second one has yields $4.23\%, \underline{4.19\%}, 4.11\%, \ldots$
* Revaluate the same portfolio with the 14 bumped curves to obtain 14 PVs
* These 14 PVs minus the base PV give you 14 PV differences, which form the Delta ladder

In [9]:
table('yc_10022025').T

1 Mo 1.5 Mo   2 Mo   3 Mo   4 Mo   6 Mo   1 Yr   2 Yr   3 Yr  \
10/2/2025  4.23%  4.18%  4.11%  4.02%  3.96%  3.81%  3.62%  3.55%  3.56%   

            5 Yr   7 Yr  10 Yr  20 Yr  30 Yr  
10/2/2025  3.67%  3.86%  4.10%  4.66%  4.69%

## Delta Ladder of a ZCB: Example

* Assuming 10y yield is $4.02\%$, what's the Delta ladder of the 10y ZCB paying $\$1$ at maturity?
* First look at the Delta at the 10y bucket. What happens to the ZCB price if 10y yield is up 1bp? 

* ZCB price today is $e^{-TR} = e^{-10\times 0.0402} = 0.66898$
* If the 10y yield is up $\Delta = 1$bp, the bond price change is $e^{-T(R+\Delta)} - e^{-TR}$
* But for small yield change we have the approximation
$$
\frac{e^{-T(R+\Delta)} - e^{-TR}}{\Delta} \approx -Te^{-TR}
$$
* Thus we have the bond price change (which is the Delta at the 10y bucket)
\begin{align*}
e^{-T(R+\Delta)} - e^{-TR} &\approx -Te^{-TR}\Delta \\
&= -10\times 0.66898\times 0.0001\\
&= -0.00066898
\end{align*}

* Yield change at other points won't affect the ZCB price a bit, so the Delta ladder is 0 at all buckets except 10y where the Delta is $-0.00066898$

## Delta Ladder of a ZCB: Example (Cont.)

* Recall that
    * A bond's price change is approximately $-D^{\$}\Delta$ for small yield change $\Delta$
    * A ZCB's dollar duration $T\times\text{(current bond price)} = T e^{-TR}$
    * Thus ZCB price change is $-Te^{-TR}\Delta$
* We've just derived the same conclusion as Chapter 4 (and if you check Chapter 4 carefully you will find it's the same derivation)
* If the term happens to be one of the curve point, the Delta ladder of a ZCB is 0 at all buckets except at the term where the Delta is $-Te^{-TR}\Delta$

## Delta Ladder of a ZCB: Exercise

* Assuming 2y yield is $3.46\%$, what's the Delta ladder of the 10y ZCB paying $\$1$ at maturity?

## Delta Ladder and PnL Computation

* Also known as the bucketed Delta, or index Delta

In [16]:
table('delta_ladder').T

1m  1.5m   2m   3m   4m   6m    1y    2y   3y  5y  7y  10y  20y  30y  \
Delta -185  -231 -303 -275 -690  778 -3739 -2320  314 -65  23   -6    1    0   

        Sum  
Delta -6698

* The risk in the above example is concentrated at at 1-2y part of the curve
* Profit and loss (PnL) computation:
    * Scenario 1: The 1m point of the curve is up 1 bp, while all other points unchanged
        * We will lose $\$185$ on the portfolio
    * Scenario 2: The 6m point is up 1 bp, while all other points unchanged
        * We will make a profit of $\$778$ on the portfolio
    * Scenario 3: The 1m and 6m points are both up 1 bp, while all other points unchanged
        * Portfolio PnL will be $778-185 = \$593$

## Delta Ladder and PnL Computation (Cont.)

In [16]:
table('delta_ladder').T

1m  1.5m   2m   3m   4m   6m    1y    2y   3y  5y  7y  10y  20y  30y  \
Delta -185  -231 -303 -275 -690  778 -3739 -2320  314 -65  23   -6    1    0   

        Sum  
Delta -6698

* PnL computation:
    * Scenario 4: The 1m point of the curve is up 2 bps, while all other points unchanged
        * We will lose $185\times 2 = \$370$ on the portfolio
    * Scenario 5: The 6m points is both up 1 bp and 1m point is up 2 bps, while all other points unchanged
        * Portfolio PnL will be $778-370 = \$408$

## Delta Ladder and Dollar Duration

In [16]:
table('delta_ladder').T

1m  1.5m   2m   3m   4m   6m    1y    2y   3y  5y  7y  10y  20y  30y  \
Delta -185  -231 -303 -275 -690  778 -3739 -2320  314 -65  23   -6    1    0   

        Sum  
Delta -6698

* PnL computation:
    * Scenario 6: Curve parallel shifts up by 1 bp
        * Portfolio PnL will be $-185-231-303 \cdots -6 + 1 + 0 = -6698$, sum of bucketed delta
* Recall that for small parallel curve shift $\Delta$, we have 
$$
-D^{\$} \Delta \approx P(\Delta) - P(0)
$$
* Thus, a bond portfolio's negative dollar duration $-D^{\$}$ should be always close to its sum of bucketed delta times $10,000 = 1/(1 \text{ bp}) = 1/\Delta$, a conclusion consistent with the previous example and exercise

## Delta Ladder and PnL Computation: Exercise

* Please compute portfolio PnL on 10/3/2025 given the below information

In [3]:
table('yc_10022025_10032025')

10/2/2025 10/3/2025  Rates Movement (bps)  Portfolio Delta Ladder
1 Mo       4.23%     4.24%                     1                    -185
1.5 Mo     4.18%     4.17%                    -1                    -231
2 Mo       4.11%     4.11%                     0                    -303
3 Mo       4.02%     4.03%                     1                    -275
4 Mo       3.96%     3.96%                     0                    -690
6 Mo       3.81%     3.82%                     1                     778
1 Yr       3.62%     3.64%                     2                   -3739
2 Yr       3.55%     3.58%                     3                   -2320
3 Yr       3.56%     3.59%                     3                     314
5 Yr       3.67%     3.72%                     5                     -65
7 Yr       3.86%     3.90%                     4                      23
10 Yr      4.10%     4.13%                     3                      -6
20 Yr      4.66%     4.69%                     3                       1
30 Yr      4.69%     4.71%                     2                       0

## Delta Ladder and PnL Computation: Answer

* PnL is the rate movement times delta, bucket by bucket, and sum up the products
$$
(1)\times(-185) + (-1)\times(-231) + (0)\times(-303) + \cdots + (3)\times(1) + 2\times0 = -\$13,195
$$